opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java

```diff
@@ -150,14 +188,57 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 IndividualProfileRemoveActivity.startIndividualProfileActivity(ChildProfileActivity.this, presenter().getChildClient(),
                         ((ChildProfilePresenter) presenter()).getFamilyID()
                         , ((ChildProfilePresenter) presenter()).getFamilyHeadID(), ((ChildProfilePresenter) presenter()).getPrimaryCareGiverID(), ChildRegisterActivity.class.getCanonicalName());
-
                 return true;
+            case R.id.action_thinkmd_health_assessment:
+                ChildFHIRBundleDao fhirBundleDao = new ChildFHIRBundleDao();
+                FHIRBundleModel bundle = fhirBundleDao.fetchFHIRDateModel(this, childBaseEntityId);
+                bundle.setEndPointPackageName(getClass().getName());
+                addThinkmdIdentifier(bundle.getUniqueIdGeneratedForThinkMD());
+                ThinkMDLibrary.getInstance().processHealthAssessment(bundle);
+                break;
```

Comment:
Can this be on the presenter and probably in a background thread

Reply:
Done